csharp/src/Client/AdbcConnection.cs

```diff
@@ -237,7 +259,25 @@ private void SetConnectionProperties(string value)
                 object? builderValue = builder[key];
                 if (builderValue != null)
                 {
-                    this.adbcConnectionParameters.Add(key, Convert.ToString(builderValue)!);
+                    string paramValue = Convert.ToString(builderValue)!;
+
+                    this.adbcConnectionParameters.Add(key, paramValue);
+
+                    switch (key)
+                    {
+                        case ConnectionStringKeywords.DecimalBehavior:
+                            this.DecimalBehavior = (DecimalBehavior)Enum.Parse(typeof(DecimalBehavior), paramValue);
+                            break;
+                        case ConnectionStringKeywords.StructBehavior:
+                            this.StructBehavior = (StructBehavior)Enum.Parse(typeof(StructBehavior), paramValue);
+                            break;
+                        case ConnectionStringKeywords.CommandTimeout:
+                            CommandTimeoutValue = ConnectionStringParser.ParseTimeoutValue(paramValue);
+                            break;
+                        case ConnectionStringKeywords.ConnectionTimeout:
+                            this.connectionTimeoutValue = ConnectionStringParser.ParseTimeoutValue(paramValue);
```

Comment:
Since this has no effect on the underlying ADBC driver, is this to help synchronize the value with the driver?

Reply:
You are right in that it doesn't do much. The initial driver value would get set when the properties are passed in for the driver. However, ConnectionTimeout is only a get call, and it would differ from the actual value that was passed in. That said, maybe this could be better if you have an idea.

Reply:
I made a slight tweak to it so that it will set the property on the driver if it is passed in and it will also have a matching get value in that case.